code/jobs/DeployJob.php

```diff
@@ -80,6 +80,22 @@ public function perform() {
 		$deployment = DNDeployment::get()->byID($this->args['deploymentID']);
 		$environment = $deployment->Environment();
 		$project = $environment->Project();
+		$backupDataTransfer = null;
+		$backupMode = !empty($this->args['backup_mode']) ? $this->args['backup_mode'] : 'db';
+
+		if(!empty($this->args['backup_before_deploy'])) {
+			$backupDataTransfer = DNDataTransfer::create();
+			$backupDataTransfer->EnvironmentID = $environment->ID;
+			$backupDataTransfer->Direction = 'get';
+			$backupDataTransfer->Mode = $backupMode;
+			$backupDataTransfer->ResqueToken = $deployment->ResqueToken;
+			$backupDataTransfer->AuthorID = $deployment->DeployerID;
+			$backupDataTransfer->write();
+
+			$deployment->BackupDataTransferID = $backupDataTransfer->ID;
```

Comment:
How about we comment the reason why we trigger a job from a job so we don't have to reverse engineer the reasoning next time around? Might make sense to write an essay next to the $db definition of the BackupDataTransferID.